restapi/src/main/java/io/stargate/web/RestApiActivator.java

```diff
@@ -53,16 +53,17 @@ public RestApiActivator() {
 
   @Override
   protected ServiceAndProperties createService() {
-    web.setAuthenticationService(authenticationService.get());
-    web.setMetrics(metrics.get());
-    web.setHttpMetricsTagProvider(httpTagProvider.get());
-    web.setAuthorizationService(authorizationService.get());
-    web.setDataStoreFactory(dataStoreFactory.get());
+    runner.setAuthenticationService(authenticationService.get());
+    runner.setMetrics(metrics.get());
+    runner.setHttpMetricsTagProvider(httpTagProvider.get());
+    runner.setAuthorizationService(authorizationService.get());
+    runner.setDataStoreFactory(dataStoreFactory.get());
     try {
-      this.web.start();
+      runner.start();
     } catch (Exception e) {
-      logger.error("Failed", e);
+      logger.error("Running RestApiRunner Failed", e);
     }
+    // Shouldn't we return something to avoid add logging for "service null"?
```

Comment:
this looks odd, but it is specified in the java docs.. It's even more confusing that `BaseActivator` does not check for returned `null` here and creates a list with a null value:

```
  protected List<ServiceAndProperties> createServices() {
    return Collections.singletonList(createService());
  }
```

So I would surely fix that `BaseActivator` stuff and then you can also just rename this method to createServices() and return empty list if you feel this is cleaner.

Reply:
Ok. I'll keep this in mind and return to it later since I suspect same pattern used elsewhere.